src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -385,12 +339,20 @@ public override void ImportParameters(RSAParameters parameters)
                 }
             }
 
+            if (!Interop.Crypto.EvpPkeySetRsa(pkey, key))
+            {
+                pkey.Dispose();
+                key.Dispose();
+                throw Interop.Crypto.CreateOpenSslCryptographicException();
+            }
+
+            key.Dispose();
```

Comment:
isn't FreeKey below already disposing the key?

Reply:
Yes, and no :smile:.

```C#
// Create the RSA object (ref count is 1)
SafeRsaHandle key = Interop.Crypto.RsaCreate();
// Create the EVP_PKEY wrapper object (ref count is 1, there's no internal key)
SafeEvpPKeyHandle pkey = Interop.Crypto.EvpPkeyCreate();

// If this succeeds, the RSA refcount goes to 2.  If it fails, it stays 1.
if (!Interop.Crypto.EvpPkeySetRsa(pkey, key))
{
    // Failure path.
    // Drop EVP_PKEY refcount to 0 and free it
    pkey.Dispose();
    // Drop RSA refcount to 0 and free it
    key.Dispose();
}

// Drop RSA refcount to 1.  The only reference now is via EVP_PKEY
key.Dispose();

...

// later, in FreeKey():
// Drop the EVP_PKEY refcount to 0, as part of its self-free it downrefs the RSA to 0
// the RSA object then frees itself.
pkey.Dispose();
```